### PR TITLE
IPv6 integration tests

### PR DESCRIPTION
**Pull request by achevuru**
**What type of PR is this?**
feature tests

**What does this PR do / Why do we need it**:
PR adds integration tests for IPv6 support.

**Testing done on this change**:
Verified the `ipv6` integration test suite on an IPv6 cluster.

**Automation added to e2e**:
Yes. `test/integration-new/cni/ipv6`

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
N/A

**Does this change require updates to the CNI daemonset config files to work?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/agent/cmd/networking/tester/network.go [hunk 1]
```diff
@@ -181,17 +189,27 @@ func TestNetworkingSetupForRegularPod(podNetworkingValidationInput input.PodNetw
...
 		log.Fatalf("failed to list ip rules %v", err)
 	}
 
 	var validationError []error
+	var podIP string
 
 	for _, pod := range podNetworkingValidationInput.PodList {
-
-		podIP, podIPNet, err := net.ParseCIDR(pod.PodIPv4Address + "/32")
+        podIP =  pod.PodIPv4Address
```
Comment (jayanthvn) [inline test/agent/cmd/networking/tester/network.go:208]: Nit: Looks like indentation is bit off.

File: test/agent/cmd/traffic-server/main.go [hunk 2]
```diff
@@ -40,8 +41,10 @@ func main() {
 		StartTCPServer(addr)
 	} else if serverMode == "udp" {
 		StartUDPServer(addr)
+	} else if serverMode == "http" {
```
Comment (jayanthvn) [inline test/agent/cmd/traffic-server/main.go:44]: Nit: Need to update "// Supported modes are tcp, udp" for var serverMode string on line 29

File: test/integration-new/ipv6/ipv6_host_networking_test.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package ipv6
```
Comment (jayanthvn) [inline test/integration-new/ipv6/ipv6_host_networking_test.go:14]: Maybe as a follow up can we also add test cases around the new env variables? ENABLE IPV4 and IPV6?

Comment (achevuru) [reply test/integration-new/ipv6/ipv6_host_networking_test.go:14]: They are covered by UT though. In general we can't toggle these 2 env variables on a live cluster as the cluster can either be an IPv4 or an IPv6 cluster.

**Review by jayanthvn — APPROVED** [review]
lgtm

**Pull request merged**